unittests/v1.2.0/test_report_generation.py

```diff
@@ -0,0 +1,87 @@
+from compliance_suite.report_runner import *
```

Comment:
Here is a suggestion to parameterize the test using different input values
```
import json
from compliance_suite.report_runner import report_runner
from parameterized import parameterized

def mock_report_runner():
    """
    This method generates and returns a list of tuples representing the good mock server report

    Each tuple contains two values:
    - authtype: Represents the authentication type.
    - good_mock_actual_report_json: Represents the report JSON generated by the mock report runner.
    """

    reports = []
    auth_types = ["none", "basic", "bearer", "passport"]
    port_numbers = ["8089", "8090", "8091", "8092"]

    for authtype, port in zip(auth_types, port_numbers):
        good_mock_actual_report = report_runner(
            server_base_url="http://localhost:" + port + "/ga4gh/drs/v1",
            platform_name="good mock server",
            platform_description=authtype,
            drs_version="1.2.0",
            config_file="compliance_suite/config/config_samples/config_" + authtype + ".json"
        )

        good_mock_actual_report_json = json.loads(good_mock_actual_report.to_json())

        # Remove timestamps
        good_mock_actual_report_json["start_time"] = ""
        good_mock_actual_report_json["end_time"] = ""
        for phase in good_mock_actual_report_json["phases"]:
            phase["start_time"] = ""
            phase["end_time"] = ""
            for test in phase["tests"]:
                test["start_time"] = ""
                test["end_time"] = ""
                for case in test["cases"]:
                    case["start_time"] = ""
                    case["end_time"] = ""

        reports.append((authtype, good_mock_actual_report_json))

    return reports

@parameterized.expand(mock_report_runner())
def test_report_runner_using_gms(authtype, good_mock_actual_report_json):
    expect_final_json = json.loads(open(f"unittests/output/expected_{authtype}.json", "r").read())
    assert good_mock_actual_report_json == expect_final_json
```

Reply:
I unfortunately do not have the bandwidth to write unit tests for the "send_request" method. However, I highly advise you to refer to a tutorial on python unit testing if you wish to proceed with the task. [Here](https://www.digitalocean.com/community/tutorials/how-to-use-unittest-to-write-a-test-case-for-a-function-in-python) is a tutorial I recommend.